test/Dialect/FIRRTL/advanced-layer-sink.mlir

```diff
@@ -817,3 +817,22 @@ firrtl.circuit "SinkRefs" {
     }
   }
 }
+
+firrtl.circuit "SinkRWProbe" {
+  firrtl.layer @A bind {}
+
+  firrtl.module public @SinkRWProbe(out %p: !firrtl.rwprobe<uint<1>, @A>) {
+    %w = firrtl.wire sym @sym : !firrtl.uint<1>
+    %0 = firrtl.ref.rwprobe <@SinkRWProbe::@sym> : !firrtl.rwprobe<uint<1>>
+    %1 = firrtl.ref.cast %0 : (!firrtl.rwprobe<uint<1>>) -> !firrtl.rwprobe<uint<1>, @A>
+    // CHECK:      %w = firrtl.wire sym @sym : !firrtl.uint<1>
+    // CHECK-NEXT: firrtl.layerblock @A {
+    // CHECK-NEXT:   %0 = firrtl.ref.rwprobe <@SinkRWProbe::@sym> : !firrtl.rwprobe<uint<1>>
+    // CHECK-NEXT:   %1 = firrtl.ref.cast %0 : (!firrtl.rwprobe<uint<1>>) -> !firrtl.rwprobe<uint<1>, @A>
+    // CHECK-NEXT:   firrtl.ref.define %p, %1 : !firrtl.rwprobe<uint<1>, @A>
+    // CHECK-NEXT: }
```

Comment:
Not for this PR: it seems like _not_ sinking `%w` is a missed optimization here given that the only user of the wire is (now) inside the layer block. This is probably worth looking into in a follow-on.

Reply:
Forceable would help with this, if we aren't removing it (it's presently unused, see https://github.com/llvm/circt/pull/7138 which removes it).